tpm2/platform.go

```diff
@@ -63,8 +83,21 @@ func (h *platformKeyDataHandler) recoverKeysCommon(data *secboot.PlatformKeyData
 			Type: secboot.PlatformHandlerErrorInvalidData,
 			Err:  fmt.Errorf("invalid key data version: %d", k.data.Version())}
 	}
+	if !k.data.Public().NameAlg.IsValid() {
+		return nil, &secboot.PlatformHandlerError{
+			Type: secboot.PlatformHandlerErrorInvalidData,
+			Err:  fmt.Errorf("invalid name algorithm for sealed object: %d", k.data.Version())}
+	}
 
-	symKey, err := k.unsealDataFromTPM(tpm.TPMContext, tpm.HmacSession())
+	var authValue []byte
+	if authKey != nil {
```

Comment:
why not `len(authKey) > 0` similar to calling spots below for consistency?

Reply:
I think this is considering nil as not providing an authKey vs possibly an empty key, though I don't know if we have empty key tests?

Reply:
I think checking `len(authKey) > 0` is fine - it is passed nil when no passphrase is set, but it's always passed a key with a length greater than zero when a passphrase is set.